utils/michigan.py

```diff
@@ -0,0 +1,1052 @@
+        for col_name in column_names:
+            non_null_values = merged_campaign_dataframe[
+                merged_campaign_dataframe[col_name].notnull()
+            ][col_name]
+
+            ids = {value: str(uuid.uuid4()) for value in non_null_values}
+
+            # Map the generated UUIDs to a new column in the DataFrame
+            merged_campaign_dataframe[
+                "{}_uuid".format(col_name)
+            ] = merged_campaign_dataframe[col_name].map(ids)
+
+        # create transaction ID for each row of the dataframe
+        merged_campaign_dataframe["transaction_id"] = [
+            uuid.uuid4() for _ in range(len(merged_campaign_dataframe))
+        ]
+
+        return merged_campaign_dataframe
+
+    def create_tables(
+        self, data: list[pd.DataFrame]
+    ) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+        """
+        Creates the Individuals, Organizations, and Transactions tables from
+        the dataframe list outputted from standardize
+
+        Inputs:
+            data: a list of 1 or 3 dataframes as outputted from standardize method.
+
+        Returns: (individuals_table, organizations_table, transactions_table)
+                    tuple containing the tables as defined in database schema
+        """
+        individuals_table, individuals_id_mapping = self.create_individuals_table(data)
+        organizations_table, organizations_id_mapping = self.create_organizations_table(
+            data
+        )
+        transactions_table, transactions_id_mapping = self.create_transactions_table(
+            data
+        )
+
+        self.output_id_mapping(
+            individuals_id_mapping, organizations_id_mapping, transactions_id_mapping
+        )
+
+        return (individuals_table, organizations_table, transactions_table)
+
+    # NOTE: The helper functions for ID_mapping output are below
+
+    def output_id_mapping(
+        self,
+        individuals_map: pd.DataFrame,
+        organizations_map: pd.DataFrame,
+        transactions_map: pd.DataFrame,
+    ) -> None:
+        """Creates MichiganIDMAp.csv
+
+        Inputs:
+            individuals_map: dataframe of individuals mapped to database
+            and provided uuid
+            organizations_map: dataframe of organizations mapped to database
+            and provided uuid
+            transactions_map: database of transactions mapped to databasee
+            and provided uuid
+
+        Returns: None, Creates data/output/MichiganIDMAp.csv
+        """
+        output_path = BASE_FILEPATH / "output" / "MichiganIDMap.csv"
+
+        michigan_id_map = pd.concat(
+            [individuals_map, organizations_map, transactions_map], ignore_index=True
+        )
+
+        if not os.path.exists(output_path.parent):
+            os.makedirs(output_path.parent)
+
+        michigan_id_map.to_csv(output_path, index=False)
+
+    def create_individuals_id_mapping(
+        self, individuals: pd.DataFrame, candidates: pd.DataFrame
```

Comment:
very confused why this is split out

Reply:
Reference to the other comment above.

 In the dataframe individual donors and organizational donors are in the same columns (first_name, last_name) etc the only way to tell them apart is if the first_name is null. The candidates data is only in the expenditures column Which is why this has to be broken out into individuals and candidates then concatenated. Adiditionally, to create the id_mapping in the structure that yall wanted I had to create this function